src/main/java/com/example/onlinebookstore/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,61 @@
+package com.example.onlinebookstore.service.impl;
+
+import com.example.onlinebookstore.dto.category.CategoryDto;
+import com.example.onlinebookstore.mapper.CategoryMapper;
+import com.example.onlinebookstore.model.Category;
+import com.example.onlinebookstore.repository.category.CategoryRepository;
+import com.example.onlinebookstore.service.CategoryService;
+import com.example.onlinebookstore.validation.category.CategoryValidator;
+import jakarta.persistence.EntityNotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryMapper categoryMapper;
+    private final CategoryValidator categoryValidator;
+    private final CategoryRepository categoryRepository;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        List<CategoryDto> categoryDtos = new ArrayList<>();
+        for (Category category : categoryRepository.findAll(pageable)) {
+            categoryDtos.add(categoryMapper.toDto(category));
+        }
+        if (categoryDtos.isEmpty()) {
+            throw new EntityNotFoundException("Cannot find any category");
+        }
+        return categoryDtos;
```

Comment:
Returning empty array with OK status is a more common and generally prefered practice for getAll() in case nothing is found
https://apihandyman.io/empty-lists-http-status-code-200-vs-204-vs-404/

Please recheck all of getAll() operations

Reply:
Thank you for advice + fixed

Reply:
Thank you for review. Fixed all changes